addons/event_sale/models/sale_order.py

```diff
@@ -83,6 +87,13 @@ def _onchange_event_ticket_id(self):
         # we call this to force update the default name
         self.product_id_change()
 
+    def unlink(self):
+        self._delete_associated_registrations()
+        super(SaleOrderLine, self).unlink()
+
+    def _delete_associated_registrations(self):
+        self.mapped('event_id.registration_ids').filtered(lambda r: r.sale_order_line_id.id in self.ids).unlink()
```

Comment:
Why not an ondelete cascade ?

Reply:
Because this is the problem I'm fixing
The `ondelete` works well, but the deletion does not trigger the method `_compute_seats`
This is the reason why I suggest to perform this ondelete on Python side, so the method will be called

Reply:
That's strange, if compute_seats is not computed this could indicate an issue ORM side. We should not hide the issue by making a workaround. Normally, unlinking a sale order unlinks its sale order lines, and then registrations, which should trigger seats computation. If that's not the case, maybe check with soemone from ORM team why this is not the case.

Reply:
[Discussion with RCO] This is a limition of the ORM: it doesn't know what has been deleted in cascade by PostgreSQL

This explains why the `compute` methods depending on `event.registration` are not triggered and why, in this case, we need to perform the cascade deletion from the ORM

Reply:
Wow, did not know that. Strange we did not have bug reports earlier then.

Instead of manually unlinking records, maybe we could simply call "_compute_seats" to event impacted by removed SO ? That way we let ondelete work and we simply add missing bit of recompute.

Reply:
Well we could have other compute, so ok to remove registrations manually, better actually.